envi/__init__.py

```diff
@@ -1528,11 +1528,20 @@ def getArchModule(name=None):
     
     return archmod
 
-def loadModuleFromPath(modname, modpathtup):
+def loadModuleFromPath(modname, modpathtup, abspath=True):
```

Comment:
You prefer having the path string building be done beforehand, and just call this with an os-specific path string?

Please explain.  This method allows for finding the module in the PYTHONPATH it using an absolute path.

Reply:
Right, but you've introduced this new parameter which I don't think we should have. I feel like we should make a call on either it being a relative path that we search the PYTHONPATH for (which presents its own weird implications about name collisions), or we mandate that it's already an absolute path and constrain how it's to be used, or we split the use cases into two functions. But I feel like this parameter overly complicates things and muddies the waters.

Reply:
I hate to just disagree, but i do.
Another function is inelegant.
We could make it always use the PYTHONPATH, but that seems artificially limiting.
It wouldn't work to use only absolute paths, drive we can't control where it could be installed.

For elegance and flexibility, i stand by my choice.

How would you change it?

Reply:
I disagree that another function is inelegant, because each function then has more clearly defined input/output and is far simpler to reason about, document, and use. Simplicity is elegance to me. 

But otherwise I would change it to one of the other two ways I mentioned. Constrain it to one of the two use cases. Of the two, I'd prefer the API to be that you pass in an absolute path. That way we head off any questions about "well I passed in X, why am I getting Y?" and it does away with some of the magic that makes people (myself included) wonder about where things are coming from. And absolute paths aren't too hard to construct. For our use case we can construct them out of something like `os.path.dirname(__file__)` to get an anchoring for where in whatever filesystem we are and build paths out of there.

Reply:
i mean, i suppose i can add a second function.  it just seems inelegant because it's yet another function name to keep track of, when a function arg makes it more flexible, and the rest of your concerns just need a better docstring.
simplicity is elegance to me, too.  but simplicity includes not having to look up which function to use.

as for `os.path.dirname(__file__)` i've always been told that's a hack and a bad idea.
i'll sleep on it.

Reply:
So I will clarify: this is not a hill I'm going to particularly die on. But it is a point of tidiness for me.

In terms of API expansion, we already have several parallels in VivWorkspace with the `loadFrom* methods, and in terms of consumers, that consistency with what we do elsewhere is valuable since we've already set the expectation. I like the parallelism, though I know it's not the only option. But the `abspath` option feels a bit hacky to me and gives me a code smell that I don't like.

I'm not sure how using `__file__` is a hack. It's been a pretty stable part of python ecosystem for a long time.